src/server/main_service.cc

```diff
@@ -1057,7 +1057,10 @@ std::optional<ErrorReply> Service::VerifyCommandState(const CommandId* cid, CmdA
       allowed_by_state = false;
       break;
     case GlobalState::TAKEN_OVER:
-      allowed_by_state = !cid->IsWriteOnly();
+      // Only Admin commands are allowed
+      // we prohibit even read commands, because read commands running in pipeline can take a while
+      // to send all data to a client which leads to fail in takeover
+      allowed_by_state = dfly_cntx.conn()->IsPrivileged();
```

Comment:
if we do this change this means that we can not takeover if replica is not connected on admin port?
is this why you needed all the changes below in replication tests?

Reply:
yes. And I thought that we do takeover always via admin port

Reply:
we do not enforce the repltakeover commands on admin port 
I am not sure about this change now as this is a breaking change
what exactly is failing ? is it the ack sent from replica or something else?

Reply:
current failed test don't connect with replication